documentation/projects/proposals/infrastructure_improvements/20240105_implementation_plan_move_remaining_services_to_openverse_org.md

```diff
@@ -0,0 +1,738 @@
+legacy root environments, so this should be a fairly easy one to move, but will
+require importing resources into the new root module and using
+`terraform state rm` to remove them from the legacy root module, in order to
+prevent them from being deleted if the legacy root module is applied.
+
+We'll move this to a new `cloudflare` root module rather than one of the `next`
+environment root modules, because there is no specific environment related to
+these settings, and it's best to keep these all together in a single place.
+Later on, in the Airflow and API work, we'll expand the Cloudflare configuration
+in `cloudflare` to include the page and WAF rules related to these services that
+are currently configured by hand in the `openverse.engineering` zone.
+
+Cloudflare rules also have priority ordering, they are matched in a specific
+order, and we need to be able to manage that ordering explicitly. We cannot
+easily (reliably) configure environment-specific rules in a separate root
+modules with priorities that reference the other environment's rules, at least
+not without creating an artificial dependency between the two (like some shared
+super-variable file the two reference to determine priority settings). It's much
+easier to write and understand if the rules are all configured in the same
+place, with priority settings that can explicitly reference each other within
+the same module or even same file.
+
+Using a separate root module also makes it possible to quickly iterate on
+changes to the Cloudflare configuration in Terraform without introducing
+dangling changes in the `next` root modules, which are more complex, and evolve
+at a slower pacing than our Cloudflare configuration sometimes does. This will
+make it possible to open PRs with changes to the Cloudflare configuration
+recorded in Terraform, while reducing the chance that another change in
+`next/production` (for example, where we might store these otherwise) could
+overwrite them.
+
+A potential alternative to using a new `cloudflare` root module to include these
+in the `org` root module, which is separated from the `next` environment root
+modules for similar reasons: the resources it configures are almost entirely
+independent of our AWS infrastructure. That would be fine, in fact, except that
+aside from the Cloudflare Access configuration, `org` is a strange name for the
+place to store Cloudflare configurations, because they aren't related to
+Openverse's organisational practices (again, aside from Cf Access). We could
+rename `org` to something else, but I'm not sure what would include the GitHub
+repository configuration and our Cloudflare configuration. I've thought about
+this quite a bit and can't come up with a name that would be _more_ intuitive
+than a separate `cloudflare` root module. If we do that, I would also like to
+rename the `org` root module to `github`, and move the configuration in its
+`github` child module into the root module.
```

Comment:
Agreed that what you're describing doesn't belong in the `org` root module, and I would be fine with renaming `org` to `github`!

Reply:
Same, +1.